ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java

```diff
@@ -281,21 +318,27 @@ public Bytes toBytes() {
           writer.writeBytes(previous_shuffling_seed);
           writer.writeBytes(current_shuffling_seed);
           // Finality
+          writer.writeBytesList(previous_epoch_attestationsBytes);
+          writer.writeBytesList(current_epoch_attestationsBytes);
           writer.writeUInt64(previous_justified_epoch.longValue());
-          writer.writeUInt64(justified_epoch.longValue());
+          writer.writeUInt64(current_justified_epoch.longValue());
+          writer.writeBytes(previous_justified_root);
+          writer.writeBytes(current_justified_root);
           writer.writeUInt64(justification_bitfield.longValue());
           writer.writeUInt64(finalized_epoch.longValue());
+          writer.writeBytes(finalized_root);
           // Recent state
           writer.writeBytesList(latest_crosslinksBytes);
           writer.writeBytesList(latest_block_roots);
+          writer.writeBytesList(latest_state_roots);
           writer.writeBytesList(latest_active_index_roots);
           writer.writeULongIntList(
               64,
               latest_slashed_balances.stream()
                   .map(UnsignedLong::longValue)
                   .collect(Collectors.toList()));
-          writer.writeBytesList(latest_attestationBytes);
-          writer.writeBytesList(batched_block_roots);
+          writer.writeBytesList(latest_block_header.toBytes());
```

Comment:
latest_block_header is not a list.
```suggestion
          writer.writeBytes(latest_block_header.toBytes());
```

Reply:
Great catch.